orchestra/models/core/models.py

```diff
@@ -659,6 +660,31 @@ class Meta:
     activity_log = JSONField(default={'actions': []})
 
 
+class TodoQA(TodoQAMixin, BaseModel):
+    """
+    A QA check for a todo on a task.
+
+    Attributes:
+        todo (orchestra.models.Todo):
+            The given todo for which QA is done.
+        approved (boolean):
+            Whether the todo has been approved or not.
+        approval_reason (str):
+            A text description explaining why a todo was
+            approved or disapproved.
+
+    Constraints:
+        `todo` has to be unique.
+    """
+    class Meta:
+        app_label = 'orchestra'
+
+    todo = models.OneToOneField(
+        Todo, related_name='qa', on_delete=models.CASCADE)
+    approval_reason = models.TextField(null=True, blank=True)
+    approved = models.NullBooleanField(blank=True)
```

Comment:
i feel less strongly about this, but you could make this `approved_datetime` for consistency :)

Reply:
This was intentional. If I were to use `approved_datetime` there will be only two states - approved or disapproved. But if I use `NullBooleanField`, I can store three states - approved, disapproved and approval pending (Null).